Src/Coravel/Scheduling/Schedule/Cron/CronExpression.cs

```diff
@@ -44,6 +46,11 @@ public CronExpression AppendWeekDay(DayOfWeek day)
 
         public bool IsDue(DateTime time)
         {
+            if (_timeZoneInfo != null)
+            {
+                time = TimeZoneInfo.ConvertTimeFromUtc(time, _timeZoneInfo);
+            }
+            
             return this.IsMinuteDue(time)
                 && this.IsHoursDue(time)
                 && this.IsDayDue(time)
```

Comment:
`IsDue` further calls `IsWeekDayDue` and then ends up doing the conversion twice. Not sure what the results of doing ` TimeZoneInfo.ConvertTimeFromUtc` the second time (using the already converted time) would be?

I'm assuming the time would remain the same? 

I generally avoid any implicit knowledge/expectations like this in code. In this case, the time, even though being "asked" to convert twice, is only converted once (if my assumption is correct). But the fact I need to make that assumption I think leads to maybe thinking about another way to do this?

The issue is that both `IsDue` and `IsWeekDayDue` are public. Perhaps using the decorator pattern here would make adding this new logic more isolated too? (Just a thought)

Reply:
well it's a good point.
the time would remain the same, but I think it's really a good idea to have a public IsWeekDay and a internal one.

Reply:
Ya, I think we'll have to create some new tests altogether with very specific scenarios.